lib/stripe/list_object.rb

```diff
@@ -16,7 +16,13 @@ def each(&blk)
     end
 
     def retrieve(id, opts={})
-      response, opts = request(:get,"#{url}/#{CGI.escape(id)}", {}, opts)
+      if id.kind_of?(Hash) # overloaded id
```

Comment:
Thanks for the PR! (And apologies for our delay in responding.)

This is a near-duplication of this logic: https://github.com/stripe/stripe-ruby/blob/master/lib/stripe/stripe_object.rb#L14-L20

Mind factoring this out into a helper method called `normalize_id`? You can put it next to `normalize_opts` in lib/stripe/util.rb. We'll want to do what the code linked above does and call `dup` on `id` before calling `delete` so we don't modify the caller's data. I like your way of setting id as the result of `delete` (rather than in a separate statement), let's keep that.